src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,733 @@
+
+        isSet = delayInfo.delay != 0 || pendingInEffect;
+
+        if (pendingInEffect) return (isSet, delayInfo.pendingDelay);
+
+        return (isSet, delayInfo.delay);
+    }
+
+    /**
+     * @param operator the operator to get the slashable magnitude for
+     * @param strategies the strategies to get the slashable magnitude for
+     *
+     * @return operatorSets the operator sets the operator is a member of and the current slashable magnitudes for each strategy
+     */
+    function getSlashableMagnitudes(
+        address operator,
+        IStrategy[] calldata strategies
+    ) external view returns (OperatorSet[] memory, uint64[][] memory) {
+        OperatorSet[] memory operatorSets = avsDirectory.getOperatorSetsOfOperator(operator, 0, type(uint256).max);
+        uint64[][] memory slashableMagnitudes = new uint64[][](strategies.length);
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            slashableMagnitudes[i] = new uint64[](operatorSets.length);
+            for (uint256 j = 0; j < operatorSets.length; ++j) {
+                slashableMagnitudes[i][j] = uint64(
+                    _magnitudeUpdate[operator][strategies[i]][_encodeOperatorSet(operatorSets[j])].upperLookupLinear(
+                        uint32(block.timestamp)
+                    )
+                );
+            }
+        }
+        return (operatorSets, slashableMagnitudes);
+    }
+
+    /**
+     * @notice Get the allocatable magnitude for an operator and strategy based on number of pending deallocations
+     * that could be completed at the same time. This is the sum of freeMagnitude and the sum of all pending completable deallocations.
+     * @param operator the operator to get the allocatable magnitude for
+     * @param strategy the strategy to get the allocatable magnitude for
+     */
+    function getAllocatableMagnitude(address operator, IStrategy strategy) external view returns (uint64) {
+        OperatorMagnitudeInfo storage info = operatorMagnitudeInfo[operator][strategy];
+        (uint64 freeMagnitudeToAdd,) = _getPendingFreeMagnitude({
+            operator: operator,
+            strategy: strategy,
+            numToComplete: type(uint16).max,
+            nextIndex: info.nextPendingFreeMagnitudeIndex
+        });
+        return info.freeMagnitude + freeMagnitudeToAdd;
+    }
+
+    /// @notice operator is slashable by operatorSet if currently registered OR last deregistered within 21 days
+    function isOperatorSlashable(address operator, OperatorSet memory operatorSet) public view returns (bool) {
+        bool isMember = avsDirectory.isMember(operator, operatorSet);
+        if (isMember) {
+            return true;
+        }
+        (, uint32 lastDeregisteredTimestamp) =
+            avsDirectory.operatorSetStatus(operatorSet.avs, operator, operatorSet.operatorSetId);
+        return block.timestamp < lastDeregisteredTimestamp + DEALLOCATION_DELAY;
```

Comment:
noting the edge case of equality here. looks like this is allowing slashing at the boundary of `lastDeregisteredTimestamp + DEALLOCATION_DELAY`, but I think this is inconsistent with the behavior elsewhere, where it seems to be assumed that slashing isn't allowed on the boundary.

Reply:
This has been replaced with `_getPendingMagnitudeInfo()` which is called for `slashOperator` and `modifyAllocations`. Here's my response to the other comment on equality boundaries. 
https://github.com/Layr-Labs/eigenlayer-contracts/pull/679#discussion_r1806723025